### PR TITLE
Adopt python3-pip-skeleton

### PR DESCRIPTION
**Pull request by callumforrester**
* Adopt https://github.com/DiamondLightSource/python3-pip-skeleton
* Fill out template values (description etc.)

### CODE REVIEW

**Review by gilesknap — COMMENTED** [review]
@callumforrester  Looks good - as the CI is passing I feel you are good to go!
(reviewing this has reminded me how much skeleton there is - that's a lot of files for an empty project)

Are you continuing with the python- prefix on project names?

File: src/dodal/__init__.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from importlib.metadata import version
 
-__version__ = version("python-dodal")
```
[the remark this answers is not in the document]
Comment (callumforrester) [reply src/dodal/__init__.py]: I forgot to specify the package name (was running `--existing`) which generated an error. I then re-ran with the package name, I assumed not everything was cleaned up from the first run.

File: tests/test_boilerplate_removed.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+"""
```
[the remark this answers is not in the document]
Comment (callumforrester) [reply tests/test_boilerplate_removed.py:1]: Think it's best to keep it as a placeholder until there are other test files

[the remark this answers is not in the document]
Comment (gilesknap) [reply tests/test_boilerplate_removed.py:1]: agreed. I made that comment before realising that there were no other tests.

**Review by DominicOram — APPROVED** [review]
Looks good to me, thank you!

**Review by noemifrisina — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**